smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/KafkaSource.java

```diff
@@ -129,6 +129,10 @@ public KafkaSource(Vertx vertx,
                                                     "10000")))
                             + 11_000L; // it's possible that it might expire 10 seconds before when we need it to
 
+                    // When resuming the consumer we must set the fetch amount to what it was before pausing
+                    // When partitions is greater than 1 this value will be 128 otherwise it will be 1.
+                    final int resumeFetchAmount = config.getPartitions() > 1 ? 128 : 1;
```

Comment:
Why 128?

Reply:
When `partitions` is greater then 1 the initial fetch is the `concurrency` value from `MultiMerge` which by default is 128.